rust_src/src/editfns.rs

```diff
@@ -1393,4 +1395,120 @@ pub fn delete_and_extract_region(
     }
 }
 
+fn time_arith<F: FnOnce(lisp_time, lisp_time) -> lisp_time>(
+    a: LispObject,
+    b: LispObject,
+    op: F,
+) -> Vec<LispNumber> {
+    let mut alen: c_int = 0;
+    let mut blen: c_int = 0;
+    let ta = unsafe { lisp_time_struct(a, &mut alen) };
```

Comment:
In theory, we could have `LispTime` in Rust which would implement the conversion logic right? Then the parameters could be `LispTime` instances. This would end up being:

    fn time_arith<F: FnOnce(LispTime, LispTime) -> LispTime>(
        a: LispTime,
        b: LispTime,
        op: F,
    ) -> Vec<EmacsInt> {
        let new_t = op(a, b);  // either check for overflow here or in the `to_vector` call below.
        new_t.to_vector()
    }

The `maxlen` would need to be computed and stored as part of the `op`.

Worth it? Over kill?

Reply:
I went with `into_vec` which accepts a length as we need to keep track of the length of the inputs. Whenever we finally move `LispTime` wholly into Rust, then we can start tracking that in the struct itself through conversions, but for now I think keeping it separate still makes the most sense.

Reply:
```
fn time_arith<F>(a: LispObject, b: LispObject, op: F) -> Vec<EmacsInt>
where
    F: FnOnce(LispTime, LispTime) -> LispTime
```

Puts the function definition first for the reader and the where clause explains what an 'op' is.